### PR TITLE
[4.4] Align the access checks for the unpublished articles in frontend category

### PR DESCRIPTION
**Pull request by ManuelHu**
This aligns the access checks for the `published` state and the `publish_up/down` checks to both use the given `filter.published`.

Pull Request for Issue #42452.

### Summary of Changes

The main `CategoryModel` model has the right access checks in

https://github.com/joomla/joomla-cms/blob/302ce2357021376a650968884ca6a5acc8e5aa59/components/com_content/src/Model/CategoryModel.php#L154-L159

Here, the access to the single category in question is checked the right way (i.e. by including the category id in the asset name).

But the category model does not filter the articles itself. For this, it calls into `ArticlesModel`:

https://github.com/joomla/joomla-cms/blob/302ce2357021376a650968884ca6a5acc8e5aa59/components/com_content/src/Model/CategoryModel.php#L233-L238

and passes on the `filter.published` state.

But in `ArticlesModel::getItems()`, the access check is repeated with a generic asset tag (i.e. `com_content` w/o any category information), but only for the `publish_up/down` case. In the simple `published` case, no additional access check is performed, and just the value of `filter.published` is used.

https://github.com/joomla/joomla-cms/blob/302ce2357021376a650968884ca6a5acc8e5aa59/components/com_content/src/Model/ArticlesModel.php#L492-L496

---

This patch essentially aligns the access checks for the published state and the publish_up/down checks to both use the given `filter.published`.

### Testing Instructions

1. Create a content category 
2. create articles in it
   * at least one with published state
   * one with published state and with publish_up in the future
   * and one set to be hidden
3. Create a user group and grant access to edit and edit-state *for this category only*
4. create a category list menu item in the frontend menu
5. log into the frontend with a user that is part of the group created in step 3

### Actual result BEFORE applying this Pull Request
* the user can only see two of the articles: the published one, and the hidden one.
* The article with publish_up in the future is not visible

### Expected result AFTER applying this Pull Request
* the logged-in user can see all three articles
* Note that this only changes the content of the list. The user always had the permission to view and edit the article (i.e. by "guessing" its URL), it was just not listed.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: components/com_content/src/Model/ArticlesModel.php [hunk 1]
```diff
@@ -490,7 +490,10 @@ protected function getListQuery()
         }
 
         // Filter by start and end dates.
-        if ((!$user->authorise('core.edit.state', 'com_content')) && (!$user->authorise('core.edit', 'com_content'))) {
+        if (
+            (!(\is_numeric($condition) && $condition == ContentComponent::CONDITION_UNPUBLISHED))
+            && (!(\is_array($condition) && \in_array(ContentComponent::CONDITION_UNPUBLISHED, $condition)))
```
Comment (Quy) [inline components/com_content/src/Model/ArticlesModel.php]:
```suggestion
            !(\is_numeric($condition) && $condition == ContentComponent::CONDITION_UNPUBLISHED)
            && !(\is_array($condition) && \in_array(ContentComponent::CONDITION_UNPUBLISHED, $condition))
```

**Comment by fgsw** [comment]
I have tested this item :white_check_mark: successfully on 3ed3af0e20dbff9bb4241c5beb5581c3893df311<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42694">issues.joomla.org/tracker/joomla-cms/42694</a>.</sub>

**Comment by jivebody** [comment]
I have tested this item :white_check_mark: successfully on 3ed3af0e20dbff9bb4241c5beb5581c3893df311<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42694">issues.joomla.org/tracker/joomla-cms/42694</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42694">issues.joomla.org/tracker/joomla-cms/42694</a>.</sub>

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**